Scripts/RemediationScripts/Remediate-DisableLocalAccountsForKubernetesService.ps1

```diff
@@ -0,0 +1,867 @@
+                                                                @{N='Name';E={$name}},
+                                                                @{N='Location';E={$location}},
+                                                                @{N='EnableRBAC';E={$enableRBAC}},
+                                                                @{N='AadProfile';E={$aadProfile}},
+                                                                @{N='DisableLocalAccounts';E={$disableLocalAccounts}} 
+ 
+            }
+        }
+        catch
+        {
+            Write-Host "Error occurred while fetching Kubernetes Cluster configurations. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+        }
+        
+        return($kubernetesClusters)
+    }
+
+    [PSObject] DisableLocalAccountsForKuberneteCluster([string] $subscriptionId, [string] $resourceName, [string] $resourceGroup, [string] $location, [boolean] $isAADEnabled, [Object[]] $aadClientIds)
+    {
+        $content = $null
+        try
+        {
+            $armUri = "https://management.azure.com/subscriptions/$($subscriptionId)/resourceGroups/$($resourceGroup)/providers/Microsoft.ContainerService/managedClusters/$($resourceName)?api-version=2022-02-01"
+            $headers = $this.GetAuthHeader()
+            $method = "Put"
+            
+            if($isAADEnabled)
+            {
+                $body =@'
+	            {
+                    "location": "{0}",
+                    "properties": {
+                        "disableLocalAccounts":true
+                    }
+                }
+'@
+
+                $jsonString = $body.Replace("{0}",$location)
+            }
+            else 
+            {
+                # If user wants to add local accounts Group to Kubernetes cluster, pass group object id as part of request body.
+                if (![String]::IsNullOrWhiteSpace($aadClientIds))
+                {
+                    $body =@'
+                    {
+                        "location": "{0}",
+                        "properties": {
+                            "aadProfile": {
+                                "managed": true,
+                                "adminGroupObjectIDs": ["{1}"]
+                            },
+                            "disableLocalAccounts":true
+                        }
+                    }
+'@
+
+                    $jsonString = $body.Replace("{0}",$location).Replace("{1}",$aadClientIds)
+                }
+                else
+                {
+                    $body =@'
+                    {
+                        "location": "{0}",
+                        "properties": {
+                            "aadProfile": {
+                                "managed": true
+                            },
+                            "disableLocalAccounts":true
+                        }
+                    }         
+'@
+                    $jsonString = $body.Replace("{0}",$location)
```

Comment:
Can the cluster have AAD Managed as true even if no AAD client id is added?

Reply:
And if Yes, then why are you asking for AAD Client Ids???

Reply:
Resolving as discussed in call.